File: otherLanguageHelpers/go/mqttRunner/pkg/utils/performancemeter/performanceMeter.go
package performancemeter

import (
	"mqttRunner/pkg/utils"
	"fmt"
)

const MS_IN_S = 1000.0

type PerformanceMeter struct {
	startTime           uint64
	startCnt            uint32
	endCnt              uint32
	_1_dividedby_endcnt float64
}

func Initialize(endCnt uint32) *PerformanceMeter {
	item := PerformanceMeter{sharedutils.CurrentTime(), 0, endCnt, 1.0 / float64(endCnt)}
	return &item
}

//Notifies this class that event has happened. Event can be for example, http request. Whatever we are measuring.
//This method prints the performance once endCnt has been reached.
func NotifyEvent(meter *PerformanceMeter) {
	meter.startCnt++
	if meter.startCnt > meter.endCnt {
		meter.startCnt = 0
		msPerRequest := float64(sharedutils.CurrentTime()-meter.startTime) * meter._1_dividedby_endcnt
		fmt.Printf("Events per second: %f\n", (1.0/msPerRequest)*MS_IN_S)
		//fmt.Println("Events per second: ", (1.0 / msPerRequest) * MS_IN_S)
		meter.startTime = sharedutils.CurrentTime()
	}
}

File: otherLanguageHelpers/nodejs_testapp/mqtt_benchmarker/utils/performance_meter.js


let startTime = Date.now();
let startCnt = 0;
let endCnt = 40000;
let _1_dividedby_endcnt = 1.0/endCnt;
const MS_IN_S = 1000.0;

module.exports={
    initialize:function (endcnt) {
        endCnt = endcnt;
        _1_dividedby_endcnt = 1.0/endCnt;
        startTime = Date.now();
    },
    notifyEvent:function () {
        if (startCnt++ > endCnt) {
            startCnt = 0;
            const msPerRequest = (Date.now() - startTime) * _1_dividedby_endcnt;
            console.log("Events per second: " + ((1.0 / msPerRequest) * MS_IN_S));
            startTime = Date.now();
        }
    }
};